### PR TITLE
ajy-UID2-1937-Disable-bulk-add-btn-when-no-changes

### PR DESCRIPTION
**Pull request by alex-yau-ttd**
https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/c2ae17dd-8e3b-43b1-aab2-b0e0f3e339a4

### CODE REVIEW

File: src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts [hunk 1]
```diff
@@ -74,6 +74,28 @@ export const hasUncheckedASharedType = (
   return false;
 };
 
+const hasPendingTypeChange = (
+  type: ClientType,
+  isChecked: boolean,
+  sharedTypes: ClientType[]
+): boolean =>
+  (!isChecked && sharedTypes.includes(type)) || (isChecked && !sharedTypes.includes(type));
```
Comment (jingyi-gao-ttd) [inline src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts:82]: maybe just `! (isChecked === sharedTypes.includes(type))`?

Comment (alex-yau-ttd) [reply src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts]: haha why didn't I think of that - done!

File: src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts [hunk 2]
```diff
@@ -74,6 +74,27 @@ export const hasUncheckedASharedType = (
...
+
+export const hasPendingTypeChanges = (
+  sharedTypes: ClientType[],
+  publisherChecked: boolean,
+  advertiserChecked: boolean,
+  DSPChecked: boolean,
+  dataProviderChecked: boolean
+): boolean => {
+  return (
+    hasPendingTypeChange('PUBLISHER', publisherChecked, sharedTypes) ||
```
Comment (jingyi-gao-ttd) [inline src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts:91]: nit: I wonder if adding hasPendingTypeChange function improves any readability compared to publisherChecked !== sharedTypes.includes('PUBLISHER');

Comment (alex-yau-ttd) [reply src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts]: good point, done!

**Review by jingyi-gao-ttd — APPROVED** [review]

**Pull request merged**